File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        freq = {}
        r,f = nums[0] , 1
        for i in range(1,len(nums)):
            if nums[i]==r:
                f += 1
            else:
                f -= 1
                if f == 0:
                    r, f = nums[i], 1
        return r

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int r = nums[0], f = 1;
        for (int i = 0; i < nums.length; i++) {
            f += (nums[i] == r ? 1 : -1);
            if (f == 0) {
                r = nums[i];
                f = 1;
            }
        }
        return r;
    }
}